File: leetcode/easy/225/225.ts
class MyStack {
  queue: number[];

  constructor() {
    this.queue = [];
  }

  push(item: number): void {
    this.queue.push(item);
    // rotate all elements except the newly added one
    for (let i = 0; i < this.queue.length - 1; i++) {
      this.queue.push(this.queue.shift()!);
    }
  }

  pop(): number | undefined {
    return this.queue.shift();
  }

  top(): number | undefined {
    return this.queue[0];
  }

  empty(): boolean {
    return this.queue.length === 0;
  }
}

File: leetcode/easy/225/225.go
package main

type MyStack struct {
	queue []int
}

func Constructor() MyStack {
	return MyStack{queue: []int{}}
}

func (this *MyStack) Push (x int) {
	this.queue = append(this.queue, x)
	for i := 0; i < len(this.queue) - 1; i++ {
		this.queue = append(this.queue, this.queue[0])
		this.queue = this.queue[1:]
	}
}

func (this *MyStack) Pop() int {
	val := this.queue[0]
	this.queue = this.queue[1:]
	return val
}

func (this *MyStack) Top() int {
	return this.queue[0]
}

func (this *MyStack) Empty() bool {
	return len(this.queue) == 0
}